packages/react/src/components/form/authenticating/otp.tsx

```diff
@@ -79,11 +95,19 @@ export const OTPState = ({ flowState }: Props) => {
         },
       };
 
+      if (!values["otp"] && hasError("otp")) {
+        clearError("otp");
+      }
       onChange(event as never);
     },
-    [registerField, text]
+    [clearError, hasError, registerField, text, values]
   );
 
+  const handleRetry = () => {
+    setFormState("retry");
```

Comment:
If we [get rid of the local retry state](https://github.com/slashid/javascript/pull/219/files#r1489300244), this wouldn't be needed anymore.